party_menu/party_menu.lua

```diff
@@ -0,0 +1,469 @@
+        start, end_point = 1, 1
+    elseif update_type == 'party_only' then
+        start, end_point = 2, 6
+    elseif update_type == 'both' then
+        start, end_point = 1, 6
+    end
+
+    for i = start, end_point do
+        if party[i] and party[i].zone_id == party[1].zone_id and not zone_check then
+            local debuffs_format_string = options.text_typeface .. ' ' .. tostring(options.text_size) .. 'px bold white'
+            local debuffs_format = string.format('[' .. get_status_string(i) .. ']{' .. debuffs_format_string .. '}')
+            party_display_strings[i].debuffs = debuffs_format
+        elseif party[i] and zone_check and party[i].zone_id == party[1].zone_id then
+            -- Don't want to change debuffs here
+        else
+            party_display_strings[i].debuffs = ''
+        end
+    end
+end
+update_party_status_strings('both')
+
+local party_window_states = {}
+
+local create_party_window_states = function()
+    for i = 1, 6 do
+        party_window_states[i] = {
+            style = 'chromeless',
+            x = options.global_window_state.x + outer_padding,
+            width = options.global_window_state.width - 2 * outer_padding,
+            height = row_window_height,
+            y = options.global_window_state.y + i * outer_padding + (i - 1) * row_window_height,
+        }
+    end
+end
+create_party_window_states()
+
+local update_ui_dimensions = function()
+    row_window_height = options.text_size + upper_inner_padding + lower_inner_padding
+    global_window_state.y = options.global_window_state.y - (party.alliance.party_1_count - 1) * (row_window_height + outer_padding)
+    global_window_state.x = options.global_window_state.x
+    global_window_state.width = options.global_window_state.width
+    global_window_state.height = party.alliance.party_1_count * (row_window_height + outer_padding) + outer_padding
+    for i, row_window in ipairs(party_window_states) do
+        row_window.x = options.global_window_state.x + outer_padding
+        row_window.width = options.global_window_state.width - 2 * outer_padding
+        row_window.y = global_window_state.y + outer_padding * i + row_window_height * (i - 1)
+        row_window.height = row_window_height
+    end
+end
+
+local check_target_for_row_highlighting = function()
+    local entity = target.t or target.st
+    for i = 1, 6 do
+        if entity and party[i] then
+            if entity.name == party[i].name then
+                party_window_states[i].color = ui.color.rgb(30, 30, 30)
+            else
+                party_window_states[i].color = ui.color.black
+            end
+        else
+            party_window_states[i].color = ui.color.black
+        end
+    end
+end
+
+-- Player zone out packet
+packets.incoming[0x00B]:register(function()
+    zone_out_time = os.clock()
+    zoning = true
+end)
+
+-- 0x0DF appears to change when player hp/mp/tp changes
+packets.incoming[0x0DF]:register(function()
```

Comment:
You receive 0x0DF any time the values for an alliance member (including yourself) change, but each chunk only contains information for one player. You shouldn't need to update the entire alliance when you receive one.